libmscore/note.cpp

```diff
@@ -2250,6 +2300,16 @@ void Note::setString(int val)
       rypos() = _string * spatium() * 1.5;
       }
 
+//---------------------------------------------------------
+//   setHeadScheme
+//---------------------------------------------------------
+
+void Note::setHeadScheme(NoteHead::Scheme val)
+      {
+      Q_ASSERT(int(val) >= -1 && int(val) < int(NoteHead::Scheme::HEAD_SCHEMES));
```

Comment:
What will happen if a wrong value is assigned in the release build?

Reply:
That's a rhetorical question, right?

Reply:
In fact, yes. I asked this question to suggestion two things: 
1. do not assign a value if it is not correct (it will be assigned in the release and everything will be bad). 
2. Add a conclusion to the log that the value is not correct.

For this, the `IF_ASSERT_FAILED` macro has recently appeared, it is located in the `log.h' file

Reply:
[Note::setHeadGroup()](
https://github.com/mattmcclinch/MuseScore/blob/bd64822d8fb3e7f18dd67fc4664dd6bd4e6335b3/libmscore/note.cpp#L2317-L2321) does something similar, and that is what I based this code on, although I had to allow for a possible correct value of -1.